runtime/src/lib.rs

```diff
@@ -239,15 +240,24 @@ pub trait Blob: Clone + Send + Sync + 'static {
     fn close(self) -> impl Future<Output = Result<(), Error>> + Send;
 }
 
+/// Interface to register and encode metrics.
+pub trait Metrics: Clone + Send + Sync + 'static {
+    /// Register a metric with the runtime.
+    ///
+    /// Any metric registered will automatically include the prefix of the current task.
+    fn register<N: Into<String>, H: Into<String>>(&self, name: N, help: H, metric: impl Metric);
```

Comment:
I'm still thinking through how to best assign a prefix here. Given metrics registry is typically called before a task is "spawned", this won't work right.

Reply:
What this looks like for `p2p` (tracked_peers is a top-level item and the things created after p2p creation have the right prefix):
<img width="504" alt="image" src="https://github.com/user-attachments/assets/df9edc7d-8c61-4835-91e0-4a4a3549115d" />

Reply:
Resolved with #517